File: code/HuaWeiNote/app/src/main/java/com/fan/hwnote/app/model/entity/Notebook.kt
package com.fan.hwnote.app.model.entity

/**
 * 笔记本实体（M12）。id == 0L 表示尚未持久化。
 *
 * folderId 必填（DB schema 同样 NOT NULL）。
 * color = 8 色调色板（spec §6）的 hex 字符串。
 * 系统预置默认笔记本 id=1, folderId=1, isDefault=true。
 */
data class Notebook(
    val id: Long = 0L,
    val name: String,
    val folderId: Long,
    val color: String = "#9E9E9E",
    val orderIndex: Int = 0,
    val isDefault: Boolean = false,
    val deletedAt: Long = 0L,
)

File: code/HuaweiNoteFlutter/lib/models/notebook.dart
class Notebook {
  final int id;
  final String name;
  final int folderId;
  final String color;
  final int orderIndex;
  final bool isDefault;
  final int deletedAt;

  const Notebook({
    required this.id,
    required this.name,
    required this.folderId,
    this.color = '#9E9E9E',
    this.orderIndex = 0,
    this.isDefault = false,
    this.deletedAt = 0,
  });
}
